File: src/main/java/org/pyjjs/scheduler/core/model/ResourceUsage.java
package org.pyjjs.scheduler.core.model;

public class ResourceUsage extends IdentifiableObject{
    private Resource resource;
    private DateRange dateRange;

    public Resource getResource() {
        return resource;
    }

    public void setResource(Resource resource) {
        this.resource = resource;
    }

    public DateRange getDateRange() {
        return dateRange;
    }

    public void setDateRange(DateRange dateRange) {
        this.dateRange = dateRange;
    }
}

File: src/main/kotlin/org/pyjjs/scheduler/core/model/ResourceUsage.kt
package org.pyjjs.scheduler.core.model

import org.pyjjs.scheduler.core.placement.time.TimePart

class ResourceUsage(var timePart: TimePart,
                    var resource: Resource,
                    var task: Task) : IdentifiableObject() {
    override fun toString(): String{
        return "ResourceUsage(timePart=$timePart, resource=$resource, task=$task)"
    }
}
